hipcheck/src/plugin/manager.rs

```diff
@@ -0,0 +1,79 @@
+use crate::hipcheck::plugin_client::PluginClient;
+use crate::plugin::{HcPluginClient, Plugin, PluginContext};
+use crate::{hc_error, Result};
+use std::process::Command;
+
+const MAX_SPAWN_ATTEMPTS: usize = 5;
+const MAX_CONN_ATTEMPTS: usize = 4;
+
+pub struct PluginExecutor {}
+impl PluginExecutor {
+	pub fn new() -> Self {
+		PluginExecutor {}
+	}
+	fn get_available_port(&mut self) -> Result<u16> {
+		for i in 40000..u16::MAX {
+			if let Ok(bind) = std::net::TcpListener::bind(format!("127.0.0.1:{i}")) {
```

Comment:
Just occurred to me that there's a TOCTOU issue with our port-assignment protocol. The RFD specifies that `hc` controls port assignment and tells the plugins what port to bind to, but that means that a port may be claimed in the time between Hipcheck checking the port's availability and a plugin binding to it.

The alternative would be to have plugins chose their own port and then tell Hipcheck about the port through an init message (perhaps this could also tie into your "ready message" idea).